### PR TITLE
Fix oculus multiplay

### PR DESCRIPTION
**Pull request by khooroko**
**Description**
Fixes OVR stuff. The tl;dr is that the original prefab had OVRManager which was singleton and spawning 2 will cause weird stuff to happen.
This PR fixes by extracting out the required stuff from OVRCameraRig and child them under something else, then delete OVRCameraRig for non-local players.

Also fixed an issue where tool fade was not synced (tools always appear faded to opponent).

Also fixed an issue where player 2 can own goal.

@lyhvictoria

**Impact**
- [x] Medium

**Test Plan**
Play 2p

### CODE REVIEW

File: Assets/Scripts/PlayerLoader.cs [hunk 1]
```diff
@@ -25,19 +25,19 @@ public class PlayerLoader : MonoBehaviour {
                 GameObject offScreenUI1 = Instantiate(OffScreenUIPrefab, OffScreenUIPrefab.transform.position, OffScreenUIPrefab.transform.rotation);
                 offScreenUI1.GetComponent<WallIndicator>().playerLocation = player1.GetComponentInChildren<Camera>(true).transform;
                 offScreenUI1.GetComponent<Canvas>().worldCamera = player1.GetComponentInChildren<Camera>();
-                player1.GetComponent<Player>().playerNumber = Utils.PlayerNumber.ONE;
+                player1.GetComponent<Player>().SetPlayerNumer(Utils.PlayerNumber.ONE);
```
Comment (lyhvictoria) [inline Assets/Scripts/PlayerLoader.cs]: spelling Number

Comment (khooroko) [reply Assets/Scripts/PlayerLoader.cs]: dammit

File: Assets/Scripts/PlayerLoader.cs [hunk 2]
```diff
@@ -25,19 +25,19 @@ public class PlayerLoader : MonoBehaviour {
...
-                    player2.GetComponent<Player>().playerNumber = Utils.PlayerNumber.TWO;
+                    player2.GetComponent<Player>().SetPlayerNumer(Utils.PlayerNumber.TWO);
                 }
             } else if (PhotonNetwork.LocalPlayer.ActorNumber == 2) {
                 player2 = PhotonNetwork.Instantiate(humanPrefab.name, player2SpawnPoint.position, player2SpawnPoint.rotation);
                 GameObject offScreenUI2 = Instantiate(OffScreenUIPrefab, OffScreenUIPrefab.transform.position, OffScreenUIPrefab.transform.rotation);
                 offScreenUI2.GetComponent<WallIndicator>().playerLocation = player2.GetComponentInChildren<Camera>(true).transform;
                 offScreenUI2.GetComponent<Canvas>().worldCamera = player2.GetComponentInChildren<Camera>();
-                player2.GetComponent<Player>().playerNumber = Utils.PlayerNumber.TWO;
+                player2.GetComponent<Player>().SetPlayerNumer(Utils.PlayerNumber.TWO);
```
Comment (lyhvictoria) [inline Assets/Scripts/PlayerLoader.cs]: spelling

File: Assets/Scripts/PlayerLoader.cs [hunk 3]
```diff
@@ -25,19 +25,19 @@ public class PlayerLoader : MonoBehaviour {
...
                 offScreenUI1.GetComponent<WallIndicator>().playerLocation = player1.GetComponentInChildren<Camera>(true).transform;
                 offScreenUI1.GetComponent<Canvas>().worldCamera = player1.GetComponentInChildren<Camera>();
-                player1.GetComponent<Player>().playerNumber = Utils.PlayerNumber.ONE;
+                player1.GetComponent<Player>().SetPlayerNumer(Utils.PlayerNumber.ONE);
                 player1.name = PhotonNetwork.LocalPlayer.NickName;
                 if (PhotonNetwork.CurrentRoom.PlayerCount == 1) {
                     // Spawn a bot if single player
                     player2 = PhotonNetwork.Instantiate(BotPrefab.name, player2SpawnPoint.position, player2SpawnPoint.rotation);
-                    player2.GetComponent<Player>().playerNumber = Utils.PlayerNumber.TWO;
+                    player2.GetComponent<Player>().SetPlayerNumer(Utils.PlayerNumber.TWO);
```
Comment (lyhvictoria) [inline Assets/Scripts/PlayerLoader.cs]: spelling

**Review by lyhvictoria — APPROVED** [review]
Pls change spelling mistakes b4 merging

**Pull request merged**